server/text_generation_server/models/paged_causal_lm.py

```diff
@@ -323,10 +323,9 @@ def __init__(
         if SPECULATOR_PATH is not None:
             from fms_extras.models.speculator import MLPSpeculator
             print(f"Speculation will be enabled up to batch size {SPECULATOR_MAX_BATCH_SIZE}")
-            self.speculator = MLPSpeculator(model_config.hidden_size, vocab_size=model_config.vocab_size, n_predict=3).to(device=self.device, dtype=dtype)
-            self.speculator.load_state_dict(
-                torch.load(SPECULATOR_PATH, map_location=self.device)["model_state"]
-            )
+            self.speculator = MLPSpeculator(model_config.hidden_size, inner_dim=4096, vocab_size=model_config.vocab_size, n_predict=4).to(device=self.device, dtype=dtype)
```

Comment:
We seem to be changing the hard-coded value for `n_predict` here and introducing another hard-coded value for `inner_dim`. How should we set these variable generally?

Reply:
I believe this speculator will have a PretrainedConfig associated with it, so we will include those there, then when you get the speculator, you can get all of the hardcoded params from the config, but til we have that, we may want to get it through some env variable? These values are different for llama and gpt bigcode